crypto/crypto_aesctr.h

```diff
@@ -8,6 +8,12 @@
 struct crypto_aes_key;
 struct crypto_aesctr;
 
+/**
+ * crypto_aes_use_x86_aesni(void):
```

Comment:
Wrong name in the comment.  Also, why expose this?  When would a caller want to use this instead of calling `crypto_aes_use_x86_aesni` directly?

... for that matter, why have it at all?  Unless there's a significant performance benefit from allowing the compiler to inline it and reference a translation-unit-local variable?

Reply:
Why use it instead of `crypto_aes_use_x86_aesni`?:
- if we didn't have `crypto_aes_key_expand()`, this would allow a caller (like `tests/crypto_aesctr/main.c`) to only `#include "crypto_aesctr.h"`.  Granted, since we're almost certain to be calling `crypto_aes_key_expand()` in the same file as the aesctr stuff, this isn't a very motivating reason.

- Performance: on my server, with `clang-11 -O2`, keeping it translation-unit-local variable avoids a penalty of between -2.7% to -0.05% (i.e. not noticeable with a buffer size of 1024).  I could benchmark it on more systems / compilers if you want.

  ![use-aes](https://user-images.githubusercontent.com/222157/103471632-5afcf780-4d37-11eb-91fb-359894633a91.png)

Reply:
Thanks, that graph is enough to convince me that we want to "cache" the value inside that translation unit.

I'm still inclined to not expose it publicly though.  Callers really shouldn't be aware of which AES-CTR code we're going to decide to use anyway (testing notwithstanding) and if they do need to know they can always include the extra header to call the AES function.